crypto/s2n_rsa.c

```diff
@@ -273,3 +208,60 @@ int s2n_rsa_keys_match(struct s2n_rsa_public_key *pub, struct s2n_rsa_private_ke
 
     return 0;
 }
+
+static int s2n_rsa_key_free(struct s2n_pkey *pkey)
+{
+    struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
+
+    RSA_free(rsa_key->rsa);
+    rsa_key->rsa = NULL;
+    
+    return 0;
+}
+
+int s2n_rsa_check_key(const struct s2n_pkey *pkey)
+{
+    const struct s2n_rsa_key *rsa_key = &pkey->key.rsa_key;
+    notnull_check(rsa_key->rsa);
+    GUARD(RSA_check_key(rsa_key->rsa));
+    return 0;
+}
+
+int s2n_evp_pkey_to_rsa_public_key(s2n_rsa_public_key *rsa_key, EVP_PKEY *evp_public_key)
+{
+    RSA *rsa = EVP_PKEY_get1_RSA(evp_public_key);
+    if (rsa == NULL) {
+        S2N_ERROR(S2N_ERR_DECODE_CERTIFICATE);
+    }
+    
+    rsa_key->rsa = rsa;
+    return 0;
+}
+
+int s2n_evp_pkey_to_rsa_private_key(s2n_rsa_private_key *rsa_key, EVP_PKEY *evp_private_key)
+{
+
+    RSA *rsa = EVP_PKEY_get1_RSA(evp_private_key);
+    if (rsa == NULL) {
+        S2N_ERROR(S2N_ERR_DECODE_PRIVATE_KEY);
+    }
+    
+    if (!RSA_check_key(rsa)) {
```

Comment:
`s2n_rsa_check_key()` should be used instead if you make the suggested change to it.